src/test/java/com/melon/portfoliomanager/AssetControllerTest.java

```diff
@@ -0,0 +1,107 @@
+package com.melon.portfoliomanager;
+
+
+import com.melon.portfoliomanager.dtos.TransactionDto;
+import com.melon.portfoliomanager.models.User;
+import com.melon.portfoliomanager.repositories.PortfolioItemRepository;
+import com.melon.portfoliomanager.repositories.TransactionRepository;
+import com.melon.portfoliomanager.repositories.UserRepository;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.system.OutputCaptureExtension;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.reactive.server.WebTestClient;
+import org.springframework.web.reactive.function.BodyInserters;
+
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ExtendWith(OutputCaptureExtension.class)
+public class AssetControllerTest {
+
+
+    @Autowired
+    private WebTestClient webTestClient;
+
+    @MockBean
+    private UserRepository userRepository;
+
+    @MockBean
+    private TransactionRepository transactionRepository;
+
+    @MockBean
+    private PortfolioItemRepository portfolioItemRepository;
+
+    private TransactionDto transactionDto;
+
+
+    @Test
+    void buyAssets_NoSuchUser_ShouldReturnNoContent() {
```

Comment:
```..._ShouldReturnNoContent``` - do we really check this scenario in this test ? Or rather we check that the controller endpoint will return 400 BadRequest ?